DiscordChatExporter.Core/Exporting/Writers/Html/MessageGroupTemplate.cshtml

```diff
@@ -81,7 +87,38 @@
 
             @{/* Right side */}
             <div class="chatlog__message-primary">
-                @if (isFirst)
+                @if (isSystemMessage)
+                {
+
+                    // system messages are grouped even if the message author is different
+                    // that's why we have to update the user values with the author of the current message
+                    userMember = Model.ExportContext.TryGetMember(message.Author.Id);
+
+                    userColor = Model.ExportContext.TryGetUserColor(message.Author.Id);
+
+                    userNick = message.Author.IsBot
+                        ? message.Author.Name
+                        : userMember?.Nick ?? message.Author.Name;                    
+                 
+                    <div class="chatlog__header">
+                        @{/* Author name */}
+                        <span class="chatlog__author" style="@(userColor is not null ? $"color: rgb({userColor.Value.R}, {userColor.Value.G}, {userColor.Value.B})" : null)" title="@message.Author.FullName" data-user-id="@message.Author.Id">@userNick</span>                        
+                    
+                        @{/* System message content */}
+                        @if(message.Kind == MessageKind.ChannelPinnedMessage)
+                        {
+                            <span class="chatlog__system-message"> pinned</span><span class="chatlog__system-message-reference-link chatlog__reference-link" onclick="scrollToMessage(event, @message.Reference.MessageId)"> a message</span><span class="chatlog__system-message"> to this channel.</span>
+                        }
+                        else
+                        {
+                            <span class="chatlog__system-message"> <!--wmm:ignore-->@Raw(FormatMarkdown(message.Content))<!--/wmm:ignore--></span>
+                        }
```

Comment:
I mean here, do an `if`/`else` for each system message kind and generate the right content. Because, technically, system messages don't have content.

Reply:
That is true, but is quite convenient actually I added some logic that makes the first letter of message.Content lowercase in the HTML export. Personally I think this is a better solution than a big `if`/`else`.

This is how it looks now:
![new](https://media.discordapp.net/attachments/554672804578983957/1019330505063272519/unknown.png)

Reply:
I agree, I think it's a good compromise now